onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -307,7 +307,7 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
       node_compute_range.Begin();
 #endif
       ORT_TRY {
-#ifdef ENABLE_TRAINING
+#if defined(ENABLE_TRAINING) || defined(ENABLE_TRAINING_OPS)
```

Comment:
Ops that need Contiguous inputs are NcclAllReduce, NcclReduceScatter, NcclAllGather... 
I don't think it need defined(ENABLE_TRAINING_OPS) here.